### PR TITLE
Update the storage adapter to handle non-anonymous users by checking …

### PR DESCRIPTION
**Pull request by knoxsp**
…whether there exists any AWS credentials in the environment

### CODE REVIEW

File: hydra_base/lib/storage/hdfstorageadapter.py [hunk 1]
```diff
@@ -74,7 +92,7 @@ def get_hdf_config(config_key="storage_hdf", **kwargs):
     @filestore_url("url")
     def open_hdf_url(self, url, **kwargs):
         try:
-            with fsspec.open(url, mode='rb', anon=True, default_fill_cache=True) as fp:
+            with fsspec.open(url, mode='rb', anon=self._get_anon(), default_fill_cache=True) as fp:
```
Comment (pmslavin) [inline hydra_base/lib/storage/hdfstorageadapter.py:95]: As the kwargs to `fsspec.open` are used several times, they can be put in a dict on the class (or prperty that returns a dict) and reused in each `fsspec.open` call, e.g.:

```python
  fsspec_args = {
    "mode": "rb",
    ...etc...
  }
  fsspec.open(url, **fsspec_args)
```

**Review by pmslavin — COMMENTED** [review]
Comment about an env var potentially relocating the credentials file.
The new credential-based access is not tested and the existing error handling is based on the assumption of anonymous access, so a test which demonstrates that...
 * credentials are used where required
 * auth errors are identified as such
...would be useful.

**Comment by pmslavin** [comment]
Regarding the issue of selecting credentials for different users and using a set of default credentials, the `s3fs` session and profile kwargs look useful for this but the documentation is quite sparse.

File: hydra_base/lib/storage/hdfstorageadapter.py [hunk 2]
```diff
@@ -58,6 +58,24 @@ def __init__(self):
...
+        """
+
+        self.accesskeyid = os.getenv('AWS_ACCESS_KEY_ID')
+        self.secretaccesskey = os.getenv('AWS_SECRET_ACCESS_KEY')
+        if self.accesskeyid is not None and self.secretaccesskey is not None:
+            return False
+
+        home = os.getenv('HOME')
+        #If there is a credentials file, then assume the user is not anonymous
+        if os.path.exists(os.path.join(home, '.aws', 'credentials')):
```
[the remark this answers is not in the document]
Comment (knoxsp) [reply hydra_base/lib/storage/hdfstorageadapter.py]: done

File: hydra_base/lib/storage/hdfstorageadapter.py (hunk 1 above)
Comment (knoxsp) [reply hydra_base/lib/storage/hdfstorageadapter.py]: done

**Comment by knoxsp** [comment]
Agreed on the s3fs profiles work, but I think that can be done in a separate pull request

**Comment by knoxsp** [comment]
Unit test added as requested

**Review by pmslavin — APPROVED** [review]
Tests and new `private_hdf_correct_access` test passing and more general AWS_SHARED_CREDENTIALS_FILE var is good.

**Pull request merged**